sunpy/map/compositemap.py

```diff
@@ -384,40 +390,28 @@ def plot(self, axes=None, annotate=True,
         # Plot layers of composite map
         for m in self._maps:
             # Parameters for plotting
-            bl = m._get_lon_lat(m.bottom_left_coord)
-            tr = m._get_lon_lat(m.top_right_coord)
-            x_range = list(u.Quantity([bl[0], tr[0]]).to(m.spatial_units[0]).value)
-            y_range = list(u.Quantity([bl[1], tr[1]]).to(m.spatial_units[1]).value)
             params = {
-                "origin": "lower",
-                "extent": x_range + y_range,
-                "cmap": m.plot_settings['cmap'],
-                "norm": m.plot_settings['norm'],
                 "alpha": m.alpha,
                 "zorder": m.zorder,
             }
             params.update(matplot_args)
 
             # The request to show a map layer rendered as a contour is indicated by a
-            # non False levels property.  If levels is False, then the layer is
-            # rendered using imshow.
+            # non False levels property.
             if m.levels is False:
-                # Check if the linewidths argument is provided, if so, then delete it from params.
-                if matplot_args.get('linewidths'):
-                    del params['linewidths']
-
-                # Check for the presence of masked map data
-                if m.mask is None:
-                    ret.append(axes.imshow(m.data, **params))
-                else:
-                    ret.append(axes.imshow(np.ma.array(np.asarray(m.data), mask=m.mask), **params))
+                # Check if any linewidth argument is provided, if so, then delete it from params.
+                for item in ['linewidth', 'linewidths', 'lw']:
+                    if item in matplot_args:
+                        del params[item]
```

Comment:
The silent deleting of these args could confuse a user trying to specify them - perhaps add a warning, or if we don't want more warnings a note in the docstring should do.

Reply:
I would be happy with a note in the docstring.

Reply:
The purpose of this deleting is so that someone can specify `.plot(linewidth=2)` and that `linewidth` argument applies only to the maps that are to be plotted as contours, because that's what a user wants.  You can't pass on the `linewidth` argument for the maps that are to be plotted as images because it does bad things.  (For imshow, it would error.  For pcolormesh, it adds an outline to each quadilateral.)

Reply:
Got it, sorry for the confusion.

Reply:
Note added to docstring